test/test/custom_activation.cpp

```diff
@@ -50,3 +50,41 @@ TEST_CASE("custom_activation")
     winrt_activation_handler = nullptr;
     clear_factory_cache();
 }
+
+// An activation handler that can only deal with Windows.Foundation.Uri.
+bool invoked_unimplemented = false;
+bool __stdcall another_handler(void* classId, winrt::guid const& iid, void** factory, int32_t* hr) noexcept
+{
+    // custom_factory is only interested in Windows.Foundation.Uri
+    if (iid != guid_of<IUriRuntimeClassFactory>()) {
+        invoked_unimplemented = true;
+        // to be explicit.
+        *factory = nullptr;
+        *hr = 0;
+        return false;
+    }
+    *factory = detach_abi(make<custom_factory>());
+    *hr = 0;
+    return true;
+}
```

Comment:
This handler is highly non-idiomatic. Should test the class name, not the interface; the fallthrough should be the failure case (in anticipation of wanting to hook multiple classes).
```cpp
std::wstring_view const name{ *reinterpret_cast<winrt::hstring*>(&classId) };
if (name == L"Windows.Foundation.Uri"v) {
    *factory = detach_abi(make<custom_factory>());
    *hr = S_OK;
    return true;
}
else
{
    // Leave unhandled
    invoked_unimplemented = true;
    return false;
}
```
This sets a good example for others who are likely to copy this code for their own `try_winrt_activation_handler` handler.

Reply:
Oops, I missed a spot in my previous recommendation. This function is noexcept, so we need to catch exceptions that come out of `make` and turn them into `hresult`s.

```cpp
bool __stdcall uri_only_activation_handler(void* classId, winrt::guid const& iid, void** factory, int32_t* hr) noexcept try
{
    ...
}
catch (...)
{
    return winrt::to_hresult();
}
```